src/subcommand/server.rs

```diff
@@ -276,7 +283,14 @@ impl Server {
         .route("/tx/:txid", get(Self::transaction))
 
         // API routes
-        .route("/rpc/v1", post(rpc::handler))
+        .route("/rpc/v1", post(rpc::handler)
+          .route_layer(TimeoutLayer::new(Duration::from_secs(self.timeout.unwrap_or(30))))
+          .route_layer(
+            ConcurrencyLimitLayer::new(
+              self.max_connections.unwrap_or(1024),
```

Comment:
Any reason for 1024? If so a comment would be nice

Reply:
don't have a very concrete reason, just from the past we have only really seen 10 qps from the api side